kong/response.go

```diff
@@ -48,14 +60,18 @@ func hasError(res *http.Response) error {
 	}
 
 	apiErr := NewAPIError(res.StatusCode, messageFromBody(body))
-	if details, ok := extractErrDetails(res); ok {
+	if details, ok := extractErrDetails(res, body); ok {
 		apiErr.SetDetails(details)
 	}
 
 	return apiErr
 }
 
-func extractErrDetails(res *http.Response) (any, bool) {
+func extractErrDetails(res *http.Response, body []byte) (any, bool) {
+	// Firstly extract `details` field from response.
```

Comment:
I think this could potentially break the expected handling of `429` from Konnect. This could happen in a situation where we get a response with `429` code and `details` filled. 

Because of that, I think that we should rather change the order here and first handle the specific cases based on the status code, and only fallback to generic details once we have no logic for a given case.

Reply:
Put extracting from body after status code process.